src/types.ts

```diff
@@ -184,6 +184,13 @@ export interface SessionRecordingOptions {
     // our settings here only support a subset of those proposed for rrweb's network capture plugin
     recordHeaders?: boolean
     recordBody?: boolean
+    /**
+     * This callback is fired when network capture has started,
+     * it can be used for example to send API calls you want to ensure
+     * are captured by PostHog that could otherwise fire on page load
+     * _before_ PostHog is ready to capture them
+     */
+    onNetworkCaptureReady?: () => void
```

Comment:
What if we do something like this: https://github.com/PostHog/posthog-js-lite/blob/e004b602463df7ef4990f2958665c622b53be252/posthog-core/src/index.ts#L153-L155
Which is a callback that can emit different types of events, then we can type an `eventType` string such as `string: onNetworkCaptureReady | captureReady | featureFlagsReady`, etc, so we can slowly add more on top of it.

Reply:
Ooh I didn't know we had that... I think the customer that requested the feature has a way forward without this PR - that's way nicer though... 🤔

Reply:
Hmmm, one for rested monday brain I think. 

posthog-js has the contract that a listener is called  when it is registered which I can't figure out with tired friday brain :)